### PR TITLE
Fsanitize fixes

### PR DESCRIPTION
**Pull request by kaleb-himes**

### CODE REVIEW

File: src/internal.c [hunk 1]
```diff
@@ -4617,6 +4617,10 @@ void SSL_ResourceFree(WOLFSSL* ssl)
 #ifdef HAVE_ED25519
     FreeKey(ssl, DYNAMIC_TYPE_ED25519, (void**)&ssl->peerEd25519Key);
     ssl->peerEd25519KeyPresent = 0;
+    #ifdef HAVE_PK_CALLBACKS
+        XFREE(ssl->buffers.peerEd25519Key.buffer, ssl->heap,
```
Comment (kaleb-himes) [inline src/internal.c]: configure option: ```./configure CC="clang -fsanitize=address" --enable-all --enable-debug```

REPORT SUMMARY:

```
=================================================================
==27102==ERROR: LeakSanitizer: detected memory leaks

Direct leak of 32 byte(s) in 1 object(s) allocated from:
    #0 0x4ca783 in malloc (wolfssl/tests/.libs/lt-unit.test+0x4ca783)
    #1 0x7f1f17c54df0 in wolfSSL_Malloc wolfssl/wolfcrypt/src/memory.c:104:15
    #2 0x7f1f17df44de in ProcessPeerCerts wolfssl/src/internal.c:8426:43
    #3 0x7f1f17f44629 in DoTls13Certificate wolfssl/src/tls13.c:4785:11
    #4 0x7f1f17f3fe70 in DoTls13HandShakeMsgType wolfssl/src/tls13.c:6123:15
    #5 0x7f1f17f4ab21 in DoTls13HandShakeMsg wolfssl/src/tls13.c:6316:15
    #6 0x7f1f17dfbc1d in ProcessReply wolfssl/src/internal.c:11591:31
    #7 0x7f1f17e8f42b in wolfSSL_connect wolfssl/src/ssl.c:8378:36
    #8 0x55e17a in client_test wolfssl/examples/client/client.c:1831:19
    #9 0x54619a in execute_test_case wolfssl/tests/suites.c:329:5
    #10 0x5443da in test_harness wolfssl/tests/suites.c:474:17
    #11 0x542cf5 in SuiteTest wolfssl/tests/suites.c:583:5
    #12 0x4f8833 in unit_test wolfssl/tests/unit.c:75:17
    #13 0x4f87b1 in main wolfssl/tests/unit.c:42:12
```

File: tests/suites.c [hunk 2]
```diff
@@ -179,6 +179,7 @@ static int execute_test_case(int svr_argc, char** svr_argv,
     THREAD_TYPE serverThread;
     char        commandLine[MAX_COMMAND_SZ];
     char        cipherSuite[MAX_SUITE_SZ+1];
+    char        portNumber[8];
```
Comment (kaleb-himes) [inline tests/suites.c]: fsanitizer reported use of variable declared in local scope after scope had ended.

File: tests/suites.c [hunk 3]
```diff
@@ -298,7 +299,6 @@ static int execute_test_case(int svr_argc, char** svr_argv,
         if (cliArgs.argc + 2 > MAX_ARGS)
             printf("cannot add the magic port number flag to client\n");
         else {
-            char portNumber[8];
```
Comment (kaleb-himes) [inline tests/suites.c]: fsanitizer reported use of variable declared in local scope after scope had ended.

File: wolfcrypt/test/test.c [hunk 4]
```diff
@@ -310,7 +310,7 @@ int memcb_test(void);
 #ifdef __cplusplus
     extern "C" {
 #endif
-    WOLFSSL_API int wolfSSL_Debugging_ON();
+    WOLFSSL_API int wolfSSL_Debugging_ON(void);
```
Comment (kaleb-himes) [inline wolfcrypt/test/test.c]: fsanitizer alerts us to old-style function definitions.

File: tests/suites.c (hunk 2 above)
Comment (dgarske) [inline tests/suites.c]: Did find one thing... Kaleb, this `portNumber` will only be used in the case `#if !defined(USE_WINDOWS_API) && !defined(WOLFSSL_TIRTOS)`. Will you add guards around that? Then its good to do.

**Comment by dgarske** [comment]
I just did a rebase -> squash merge -> force push with the portNumber fix. Once Jenkins is done it will be ready for third-party review, so assigning over to Todd.

**Pull request merged**